src/plugin.rs

```diff
@@ -95,9 +104,33 @@ macro_rules! appbase_plugin_requires {
                return false;
             }
             self.base.state = PluginState::Stopped;
+            $(appbase_plugin_requires_visit!($deps, shutdown);)*
             log::info!("plugin shutdown: {}", <$name>::typename());
             true
          }
       }
    };
 }
+
+#[macro_export]
+macro_rules! appbase_register_async_single {
+   ($self:ident; $run:block;) => {
```

Comment:
```suggestion
   ($self:ident, $run:block) => {
```
What about making macro like this?